x-pack/test/api_integration/apis/ml/trained_models/delete_model.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../../../ftr_provider_context';
+import { USER } from '../../../../functional/services/ml/security_common';
+import { COMMON_REQUEST_HEADERS } from '../../../../functional/services/ml/common_api';
+
+export default ({ getService }: FtrProviderContext) => {
+  const supertest = getService('supertestWithoutAuth');
+  const ml = getService('ml');
+
+  describe('DELETE trained_models', () => {
+    before(async () => {
+      await ml.testResources.setKibanaTimeZoneToUTC();
+      await ml.api.createdTestTrainedModels('regression', 2);
+    });
+
+    after(async () => {
+      await ml.api.cleanMlIndices();
+    });
+
+    it('deletes trained model by id', async () => {
+      const { body: deleteResponseBody } = await supertest
+        .delete(`/api/ml/trained_models/dfa_regression_model_n_0`)
+        .auth(USER.ML_POWERUSER, ml.securityCommon.getPasswordForUser(USER.ML_POWERUSER))
+        .set(COMMON_REQUEST_HEADERS)
+        .expect(200);
+
+      expect(deleteResponseBody).to.eql({ acknowledged: true });
+
+      // verify that model is actually deleted
+      await supertest
+        .get(`/api/ml/trained_models/dfa_regression_model_n_0`)
+        .auth(USER.ML_POWERUSER, ml.securityCommon.getPasswordForUser(USER.ML_POWERUSER))
+        .set(COMMON_REQUEST_HEADERS)
+        .expect(404);
+    });
+
+    it('does not allow to delete trained model if the user does not have required permissions', async () => {
+      await supertest
+        .delete(`/api/ml/trained_models/dfa_regression_model_n_0`)
+        .auth(USER.ML_VIEWER, ml.securityCommon.getPasswordForUser(USER.ML_VIEWER))
+        .set(COMMON_REQUEST_HEADERS)
+        .expect(403);
```

Comment:
Would be good to verify that the model still exists here

Reply:
Should we verify that this endpoint should not work with ML_UNAUTHORIZED as well?

Reply:
This model is actually deleted in the previous tests, the endpoint validates permission before the actual delete attempt, so we get a `403` response. 
Regarding the `ML_UNAUTHORIZED ` user, it'd be redundant as the `ML_VIEWER` has more permissions, e.g. read access, while `ML_UNAUTHORIZED ` doesn't have any.

Reply:
Added extra tests and assertion in https://github.com/elastic/kibana/pull/104819/commits/eb7f505eb2ffe3cac6d7c65b60cc5d6a6c4d39c7